openedx/features/job_board/views.py

```diff
@@ -114,9 +117,20 @@ class JobDetailView(DetailView):
     template_name = 'features/job_board/job_detail.html'
 
 
-class JobCreateView(CreateView):
+@method_decorator(ensure_csrf_cookie, name='dispatch')
+class JobCreateView(LoginRequiredMixin, CreateView):
+    """
+    Create job view and submit job posting data
+    """
     form_class = JobCreationForm
     template_name = 'features/job_board/create_job_form.html'
 
+    def get_initial(self, *args, **kwargs):  # pylint: disable=unused-argument, arguments-differ
+        """Pre-fill form with initial data"""
+        initial = super(JobCreateView, self).get_initial(**kwargs)
+        user = self.request.user
+        initial['user'] = user
```

Comment:
in this case now, we can also directly `initial['user'] = self.request.user`

Just a casual suggestion, feel free to ignore it.